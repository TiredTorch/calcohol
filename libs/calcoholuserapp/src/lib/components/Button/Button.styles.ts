export const buttonStyles = {
	root: {
		boxSizing: "border-box",
		filter: "drop-shadow(5px 7px 8px rgba(54, 123, 203, 0.25))",
		color: "#367BCB",
		width: "100%",
		maxHeight: "90px",
		height: "10vh",
		textTransform: "none",
		fontSize: "25px",
		fontWeight: "400",
		lineHeight: "30px",
		letterSpacing: "0em",
		textAlign: "center",

		background: "rgba(132, 73, 208, 0.1)",
		border: "2px solid rgba(145, 54, 54, 0.15)",
		boxShadow: "3px 7px 4px rgba(0, 0, 0, 0.25)",
		borderRadius: "10px",
	}
};