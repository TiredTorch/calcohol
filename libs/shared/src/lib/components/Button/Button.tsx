import { FC } from "react";
import { ButtonProps, Button as MUIButton } from "@mui/material";

export const Button: FC<ButtonProps> = ({
	children,
	...rest
}) => {
	return (
		<MUIButton
			{...rest}
		>
			{children}
		</MUIButton>
	);
};

export default Button;