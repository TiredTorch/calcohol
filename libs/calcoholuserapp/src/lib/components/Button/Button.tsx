import { FC } from "react";
import { Button as MUIButton, ButtonProps } from "@mui/material";
import { buttonStyles } from "./Button.styles";

export const Button: FC<ButtonProps> = ({
	children,
	sx,
	...rest
}) => {
	return (
		<MUIButton
			{...rest}
			sx={[buttonStyles.root, sx]}
		>
			{children}
		</MUIButton>
	);
};

export default Button;