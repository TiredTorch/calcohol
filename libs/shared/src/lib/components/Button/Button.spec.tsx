import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { faker } from "@faker-js/faker";
import { Button } from "./Button";

afterEach(() => {
	jest.resetAllMocks();
});

const mockedTitle = faker.datatype.string(10);

it("should render button with text", () => {
	const { baseElement } = render(
		<Button>{mockedTitle}</Button>
	);

	expect(baseElement).toBeInTheDocument();
	expect(baseElement).toMatchSnapshot();
	expect(baseElement).toHaveTextContent(mockedTitle);
});
