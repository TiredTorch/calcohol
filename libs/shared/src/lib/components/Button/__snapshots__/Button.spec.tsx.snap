// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button with text 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      D)0JO2#O0#
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
